php-redis-cache/src/RedisCacheItemPool.php

```diff
@@ -0,0 +1,222 @@
+        }, $keys);
+    }
+
+    /**
+     * Confirms if the cache contains specified cache item.
+     *
+     * Note: This method MAY avoid retrieving the cached value for performance reasons.
+     * This could result in a race condition with CacheItemInterface::get(). To avoid
+     * such situation use CacheItemInterface::isHit() instead.
+     *
+     * @param string $key
+     *   The key for which to check existence.
+     *
+     * @throws InvalidArgumentException
+     *   If the $key string is not a legal value a \Psr\Cache\InvalidArgumentException
+     *   MUST be thrown.
+     *
+     * @return bool
+     *   True if item exists in the cache, false otherwise.
+     */
+    public function hasItem($key): bool
+    {
+        if (!is_string($key)) {
+            throw new InvalidArgumentException(gettype($key), ['string']);
+        }
+
+        return (bool)$this->client->exists($key);
+    }
+
+    /**
+     * Deletes all items in the pool.
+     *
+     * @return bool
+     *   True if the pool was successfully cleared. False if there was an error.
+     */
+    public function clear(): bool
+    {
+        $this->client->flushdb();
+
+        return $this->client->dbsize() === 0;
+    }
+
+    /**
+     * Removes the item from the pool.
+     *
+     * @param string $key
+     *   The key to delete.
+     *
+     * @throws InvalidArgumentException
+     *   If the $key string is not a legal value a \Psr\Cache\InvalidArgumentException
+     *   MUST be thrown.
+     *
+     * @return bool
+     *   True if the item was successfully removed. False if there was an error.
+     */
+    public function deleteItem($key): bool
+    {
+        if (!is_string($key)) {
+            throw new InvalidArgumentException(gettype($key), ['string']);
+        }
+
+        $this->client->del($key);
+
+        return !$this->hasItem($key);
+    }
+
+    /**
+     * Removes multiple items from the pool.
+     *
+     * @param string[] $keys
+     *   An array of keys that should be removed from the pool.
+     * @throws InvalidArgumentException
+     *   If any of the keys in $keys are not a legal value a \Psr\Cache\InvalidArgumentException
+     *   MUST be thrown.
+     *
+     * @return bool
+     *   True if the items were successfully removed. False if there was an error.
+     */
+    public function deleteItems(array $keys): bool
+    {
+        $fail = false;
+
+        foreach ($keys as $key) {
+            if (!$this->deleteItem($key)) {
+                $fail = true;
+            }
+        }
+
+        return $fail ? false : true;
+    }
+
+    /**
+     * Persists a cache item immediately.
+     *
+     * @param CacheItemInterface $item
+     *   The cache item to save.
+     *
+     * @return bool
+     *   True if the item was successfully persisted. False if there was an error.
+     */
+    public function save(CacheItemInterface $item): bool
+    {
+        return $this->client->set($item->getKey(), $item->get()) === 'OK';
```

Comment:
Do you have any idea how to obtain `ttl` for `client->set(..., ..., $expireResolution, $expireTTL)`?
`CacheItemInterface` does not provide any getter for expiration...